framework/py/flwr/server/superlink/linkstate/in_memory_linkstate.py

```diff
@@ -391,29 +391,13 @@ def get_nodes(self, run_id: int) -> set[int]:
                 if info.online_until > current_time
             }
 
-    def set_node_public_key(self, node_id: int, public_key: bytes) -> None:
-        """Set `public_key` for the specified `node_id`."""
-        with self.lock:
-            if node_id not in self.nodes:
-                raise ValueError(f"Node {node_id} not found")
-
-            if public_key in self.public_key_to_node_id:
-                raise ValueError("Public key already in use")
-
-            self.public_key_to_node_id[public_key] = node_id
-            self.node_id_to_public_key[node_id] = public_key
-
     def get_node_public_key(self, node_id: int) -> Optional[bytes]:
         """Get `public_key` for the specified `node_id`."""
         with self.lock:
-            if node_id not in self.nodes:
-                raise ValueError(f"Node {node_id} not found")
```

Comment:
Should we leave this here? checking if the node exists is the first stage also in other methods.